packages/grafana-data/src/types/dataFrame.ts

```diff
@@ -98,6 +98,9 @@ export interface FieldConfig<TOptions = any> {
   // The field type may map to specific config
   type?: FieldTypeConfig;
 
+  /** @alpha Whether we should treat this field as nested data */
```

Comment:
I think field(s) that holds the nested frame values makes more sense and would support multiple levels of nesting. 

We already have support for data frame value fields. Just need to mark them as holding nested frames so table panel shows them when expanding a row (instead of as columns)

Reply:
If I'm understanding this correctly, then I think the idea is to group all the nested fields into one field, where each value in that single field can be an array of data frames, where each data frame is one of the nested tables for the row at that values (array of data frames) index. I've pushed a commit to address this.

Reply:
We can have multiple fields with nested frames, I think that is fine. 

Otherwise each field value would be an array of data frames with could work but not sure how messy that is. But might be easier than dealing with multiple fields

Reply:
> This approach uses a column/field with a DataFrame type to hold nested data

Yes, this is what I meant.

Reply:
Ah ok so multiple fields with their own nested data is the approach I went with first in this PR, but felt it was a cleaner to have one field with all the nested data and pushed updates for that. I've left it as is for the moment - we can easily compare what changed ([de0fb7e](https://github.com/grafana/grafana/pull/71953/commits/de0fb7ee8dd6dff8f4b58899f7cbb8e331b7e717)), but happy to change it back to multiple fields if necessary.